webui/react/src/pages/TrialLogs.tsx

```diff
@@ -37,6 +37,7 @@ const TrialLogs: React.FC = () => {
   const [ isDownloading, setIsDownloading ] = useState(false);
   const [ isLoading, setIsLoading ] = useState(true);
   const [ isIdInvalid, setIsIdInvalid ] = useState(false);
+  const [ downloadModal, setDownloadModal ] = useState<{ destroy: () => void }>();
```

Comment:
The code seems correct, but I don't fully understand why having a function as state is necessary here. @hkang1 can you explain to me why this is the chosen approach and why something like a `modalIsVisible` state wasn't chosen?

Reply:
Great questions!

Regarding the function comment. This portion is creating a state object with the `destroy` function shape defined for Typescript. Which is what antd `Modal.confirm` returns. It actually has other properties besides `destroy` but we only need and care about the `destroy` function which we need to call to forcibly remove the modal because the Ok interaction by default does not dismiss the modal due to it being tied to a promise based handler (which is great to have).

As you hinted, this could have been written as part of a permanent JSX modal and just toggling the visibility on and off. I guess I initially leaned more towards this approach to make the modal management more or less encapsulated in the handler functions to make it easier to remove. The idea being this modal is a temporary measure until we figure out the proper logs download method.

Reply:
Looking at it, the the effort to write it as a JSX modal is just as easy, and it would be more consistent with all the other modals we've written. One slight benefit of a dynamic modal is that resources aren't allocated for the modal until it's actually needed, but in this case the content of the modal is so trivial that it really makes no difference.

Reply:
Ahhh, that  that makes a ton more sense. Thanks!